From 62174518a09fd4c8375abc79e27acf22b344c13b Mon Sep 17 00:00:00 2001
Date: Sun, 16 May 2021 13:20:14 -0700
Subject: [PATCH 21/24] 10.6 and less: libcxx disable Availability tests

the libc++.dylib installed by MacPorts has different features than
the OS default-installed version, so the Availability tests are
not valid.

Currently this is done by default on SnowLeopard and earlier.

note this patch is disappearing. The new way to specify this will be
with a Cmake define

LIBCXX_ENABLE_VENDOR_AVAILABILITY_ANNOTATIONS
---
 libcxx/include/__config | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libcxx/include/__config b/libcxx/include/__config
index a3838c89e8e1..685827372fd5 100644
--- a/libcxx/include/__config
+++ b/libcxx/include/__config
@@ -10,6 +10,9 @@
 #ifndef _LIBCPP_CONFIG
 #define _LIBCPP_CONFIG
 
+// for MacPorts, our installed libcxx does not require Apple system Availability tests
+#define _LIBCPP_DISABLE_AVAILABILITY
+
 #if defined(_MSC_VER) && !defined(__clang__)
 #  if !defined(_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
 #    define _LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER
-- 
2.21.1 (Apple Git-122.3)

